### PR TITLE
build: Update dependencies on `@fluentui/react-components`

### PR DESCRIPTION
**Pull request by Josmithr**
Also migrates off of lingering usages of `/unstable` APIs, allowing us to move to a `^` dependency, instead of a `~`.

### CODE REVIEW

**Comment by Josmithr** [comment]
Investigating the broken test

File: packages/tools/devtools/devtools-example/package.json [hunk 1]
```diff
@@ -36,12 +36,12 @@
 		"test:jest:coverage": "jest --detectOpenHandles --coverage --ci",
 		"tsc:watch": "tsc --watch",
 		"watch:devtools-core": "tsc --watch --project ../devtools-core/tsconfig.json --preserveWatchOutput",
-		"watch:devtools-view": "tsc --watch --project ../devtools-view/tsconfig.json --preserveWatchOutput",
+		"watch:devtools-view": "tsc --watch --project ../devtools-view/tsconfig.esm.json --preserveWatchOutput",
```
Comment (jikim-msft) [inline packages/tools/devtools/devtools-example/package.json:39]: Why is this needed?

**Review by jikim-msft — APPROVED** [review]

File: packages/tools/devtools/devtools-example/package.json (hunk 1 above)
Comment (Josmithr) [reply packages/tools/devtools/devtools-example/package.json:39]: Now that we've migrated to explicit `cjs` and `esm` builds, invoking tsc with our "base" configs spits out errors (currently in the form of a bunch of bogus build output files). Since the example app is `esm`-only, we want to specify the esm build config for devtools-view.

devtools-core doesn't have the same setup (though I'm not sure why), so its config is left alone here for now.

File: packages/tools/devtools/devtools-view/src/test/jest/VisualTree.test.tsx [hunk 2]
```diff
@@ -118,15 +91,9 @@ describe("VisualTreeView component tests", () => {
 			</MessageRelayContext.Provider>,
 		);
 
-		// TODO: Loop the expand button for n-amount of times.
-		const expandButton = await screen.findByTestId("tree-button");
-		await userEvent.click(expandButton);
-
-		await screen.findByText(/test-node-key/);
```
Comment (Josmithr) [inline packages/tools/devtools/devtools-view/src/test/jest/VisualTree.test.tsx:125]: It looks like this test was already not working. This text appears regardless of the click event above. I've confirmed that the way the tree component manages expanded/collapsed content doesn't align with the testing pattern here. I've left a TODO to add interactive tests for this component. For now, I've simplified the case to simply verify that the label we're looking for exists.

**Review by alexvy86 — APPROVED** [review]
